src/overlays/actors/ovl_En_Sth2/z_en_sth2.c

```diff
@@ -27,22 +31,95 @@ const ActorInit En_Sth2_InitVars = {
     (ActorFunc)NULL,
 };
 
-#endif
+#include "overlays/ovl_En_Sth2/ovl_En_Sth2.c"
+
+static Vec3f D_80BFA010 = { 700.0f, 400.0f, 0.0f };
+
+static Color_RGB8 sEnvColors[] = { { 190, 110, 0 },  { 0, 180, 110 },   { 0, 255, 80 },
+                                   { 255, 160, 60 }, { 190, 230, 250 }, { 240, 230, 120 } };
+
+void EnSth2_Init(Actor* thisx, PlayState* play) {
+    EnSth2* this = THIS;
+
+    this->objIndex = Object_GetIndex(&play->objectCtx, OBJECT_STH);
+    Actor_SetScale(&this->actor, 0.01f);
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 36.0f);
+    this->unused = 0;
+
+    if (play->actorCtx.unk5 & 2) {
+        this->actor.flags |= (ACTOR_FLAG_10 | ACTOR_FLAG_20);
+    } else {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+    this->actionFunc = EnSth2_UpdateAnimation;
+}
+
+void EnSth2_Destroy(Actor* thisx, PlayState* play) {
+}
+
+void EnSth2_UpdateAnimation(EnSth2* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
+}
+
+void EnSth2_Update(Actor* thisx, PlayState* play) {
+    s32 pad;
+    EnSth2* this = THIS;
+
+    if (Object_IsLoaded(&play->objectCtx, this->objIndex)) {
+        this->actor.objBankIndex = this->objIndex;
+        Actor_SetObjectDependency(play, &this->actor);
+        SkelAnime_InitFlex(play, &this->skelAnime, &object_sth_Skel_0031F8, &gEnSth2WavingHandAnim, &this->jointTable,
+                           &this->morphTable, 16);
```

Comment:
```suggestion
                           &this->morphTable, OBJECT_STH_LIMB_MAX);
```
That last parameter is the number of limbs of the skeleton you are passing.